blob/blob.go

```diff
@@ -381,3 +410,12 @@ func IsNotExist(err error) bool {
 	}
 	return false
 }
+
+// IsNotImplemented returns true iff err indicates that the provider does not
+// support the given operation.
+func IsNotImplemented(err error) bool {
```

Comment:
Testing for rationale: do we expect that calling code would branch based on this information? I can't think of a concrete example here.

Reply:
Interesting question. I mostly want this to be a very clear error message, where checking it is not driver-specific string parsing. I can imagine an app trying to sign a URL on startup to see if the current provider supports it, and disabling some functionality otherwise.

Reply:
Can we move the check forward? For example, the user may found out there isn't a signed URL enabled provider when they assemble their providerset.

Reply:
That's what the [enforcing portability](https://github.com/google/go-cloud/blob/master/internal/docs/design.md#enforcing-portability) would achieve, but we decided to wait on implementing that until we had more evidence that users want it. I don't think there's a way to fail at startup/init time that doesn't involve the user telling us what features they want.